### ISSUE TITLE
cannot access ListenableFuture

### ISSUE THREAD

**Issue by Abesoddy** [thudugala/Plugin.LocalNotification#27]
Hello,

I have this error on the project Xamarin.Froms.Android with version 4.0.3 :

Error:  error: cannot access ListenableFuture public class ScheduledNotificationWorker class file for com.google.common.util.concurrent.ListenableFuture not found

Does this come from the plugin or an implementation error?

**Comment by thudugala** [comment]
Can you please attach a sample project?

**Comment by tranb3r** [comment]
@thudugala @Abesoddy 

I'm having the same build error when attempting to build the sample project (Nuget) for Android.

However, the error only appears when using visual studio preview:
- with Visual Studio 2019 16.1.4 (xamarin sdk 9.3.0.23): no error on build
- with Visual Studio 2019 16.2.0 preview 3.0 (xamarin sdk 9.4.0.4): build error (cannot access ListenableFuture).

Any idea for fixing it ? Is it related to a known issue in the VS preview ? Is there a workaround ? Or maybe this is something new that should be reported to the VS team ?

It could be worth re-opening this issue until we have a full resolution.

**Comment by tranb3r** [comment]
Any progress on that one ?

**Comment by tranb3r** [comment]
Same issue with the new preview: Visual Studio 2019 16.2.0 preview 4.0 (xamarin android sdk 9.4.0.51)

**Comment by thudugala** [comment]
@tranb3r Can you try referencing Xamarin.Android.Arch.Work.Runtime in your android project ?

**Comment by tranb3r** [comment]
> @tranb3r Can you try referencing Xamarin.Android.Arch.Work.Runtime in your android project ?

Same error (tested with the sample project (nuget)).

**Comment by pikausp** [comment]
Possibly related to https://github.com/xamarin/xamarin-android/issues/3343?

**Comment by thudugala** [comment]
https://docs.microsoft.com/en-nz/xamarin/android/release-notes/9/9.4#known-issues-in-xamarinandroid-94051

Workaround:
Add the following lines to the bottom of your app project's .csproj file (just before the closing </Project> tag):

```xml
<Target Name="FixGuava" AfterTargets="_AddAndroidCustomMetaData">
  <ItemGroup>
    <_MonoAndroidReferencePath Include="@(_ReferencePath);@(_ReferenceDependencyPaths)" Condition=" '%(Filename)' == 'Xamarin.Google.Guava.ListenableFuture' " />
  </ItemGroup>
</Target>
```

**Comment by thudugala** [comment]
Issue seems to be fixed in VS 16.2.1 (Xamarin.Android 9.4.1.0)
